activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -360,28 +372,16 @@ def references(*args, **options)
       end
       alias :belongs_to :references
 
-      def new_column_definition(name, type, options) # :nodoc:
+      def new_column_definition(name, type, **options) # :nodoc:
         type = aliased_types(type.to_s, type)
-        column = create_column_definition name, type
-
-        column.limit       = options[:limit]
-        column.precision   = options[:precision]
-        column.scale       = options[:scale]
-        column.default     = options[:default]
-        column.null        = options[:null]
-        column.first       = options[:first]
-        column.after       = options[:after]
-        column.auto_increment = options[:auto_increment]
-        column.primary_key = type == :primary_key || options[:primary_key]
-        column.collation   = options[:collation]
-        column.comment     = options[:comment]
-        column.as          = options[:as]
-        column
+        options[:primary_key] ||= type == :primary_key
+        options[:null] = false if options[:primary_key]
```

Comment:
@kamipo @jeremy  I'm wondering why this was added? This prevents setting `null: true` when combined with the `primary_key` option. e.g.

```rb
add_reference(
  :orders,
  :user,
  null: true,
  foreign_key: true,
  primary_key: :custom_id
)
```

Will output:

```sql
ALTER TABLE "orders" ADD "user_id" uuid NOT NULL
```

Even when you specified `null: true`

Maybe I'm missing something?